app/lib/azure_environment.rb

```diff
@@ -4,6 +4,8 @@ def self.authorised_hosts
   end
 
   def self.hostname
-    ENV.fetch('CUSTOM_HOSTNAME', authorised_hosts.first)
+    hostname = authorised_hosts.first rescue nil
+    hostname ||= Socket.gethostname
+    ENV.fetch('CUSTOM_HOST_NAME', hostname)
```

Comment:
This will need a rebase (another!) against #408 where we've now renamed the env var

Reply:
I like falling back to `Socket.hostname` but I feel like this code might not belong inside the `AzureEnvironment` module. I *think* I'd rather have 

```
domain: Rails.env.production? ? AzureEnvironment.hostname : Socket.hostname
```

in `LogStashLogging`? This module is then 100% about Azure and the dependency on those ENV vars is clear. What do you think?

Reply:
good call, updating